### PR TITLE
Rework DynamoTestEnv.

### PR DESCRIPTION
**Pull request by vpodlnk**
Should resolve https://github.com/PlayQ/d4s/issues/41

### CODE REVIEW

File: d4s-test/src/main/scala/d4s/test/envs/DynamoTestEnv.scala [hunk 1]
```diff
@@ -3,15 +3,27 @@ package d4s.test.envs
...
+import izumi.distage.model.definition.StandardAxis.Env
 import izumi.distage.model.definition.{DIResource, Module}
+import izumi.distage.testkit.TestConfig
+import izumi.distage.testkit.scalatest.DistageBIOSpecScalatest
 import logstage.LogBIO
 import net.playq.aws.tagging.AwsNameSpace
 import zio.{IO, Task}
 
-trait DynamoTestEnv {
+trait DynamoTestEnv extends DistageBIOSpecScalatest[IO] {
```
Comment (neko-kai) [inline d4s-test/src/main/scala/d4s/test/envs/DynamoTestEnv.scala:17]:
```scala
trait DynamoTestEnv[F[_]] { this: DistageAbstractScalatestSpec[F] }
```
We don't want to force `zio.IO`, distage-testkit works with any effect monad

File: d4s-test/src/main/scala/d4s/test/envs/DynamoTestEnv.scala [hunk 2]
```diff
@@ -3,15 +3,27 @@ package d4s.test.envs
...
+  val envTest = Activation(Env -> Env.Test)
```
Comment (neko-kai) [inline d4s-test/src/main/scala/d4s/test/envs/DynamoTestEnv.scala:18]: this is part of default `config.activation`, not necessary to override.

File: d4s-test/src/main/scala/d4s/test/envs/DynamoTestEnv.scala [hunk 3]
```diff
@@ -3,15 +3,27 @@ package d4s.test.envs
...
+
+  override protected def config: TestConfig = super.config.copy(
+    moduleOverrides  = moduleOverrides,
+    activation       = super.config.activation ++ envTest,
+    memoizationRoots = memoizationRoots,
+    forcedRoots      = additionalRoots,
+  )
+
   def moduleOverrides: Module = new ModuleDef {
```
Comment (neko-kai) [inline d4s-test/src/main/scala/d4s/test/envs/DynamoTestEnv.scala:27]: You can inline all of these into `config.copy`

Comment (vpodlnk) [reply d4s-test/src/main/scala/d4s/test/envs/DynamoTestEnv.scala]: Done ✅

File: d4s-test/src/main/scala/d4s/test/envs/DynamoTestEnv.scala (hunk 1 above)
Comment (vpodlnk) [reply d4s-test/src/main/scala/d4s/test/envs/DynamoTestEnv.scala]: Done ✅

File: d4s-test/src/main/scala/d4s/test/envs/DynamoTestEnv.scala (hunk 2 above)
Comment (vpodlnk) [reply d4s-test/src/main/scala/d4s/test/envs/DynamoTestEnv.scala]: Fixed ✅

**Pull request merged**